### PR TITLE
Existing annotations mapping

### PR DESCRIPTION
**Pull request by yolandeleungFT**

### CODE REVIEW

File: mapper/mapper.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	err := json.Unmarshal(body, &originalAnnotations)
+	if err != nil {
+		log.Fatal("Could not unmarshall json body", err)
+	}
+
+	for i := 0; i < len(originalAnnotations); i++ {
+		annoMap := originalAnnotations[i]
+		pred, ok := annoMap["predicate"]
+		if !ok {
+			log.Info("no predicate")
```
Comment (galiri) [inline mapper/mapper.go]: maybe " no predicate supplied  for annotation" , as mentioned annotations-rw-neo4j makes no predicate into MENTIONS , do we want that logic?

File: mapper/mapper.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+		stringTypes, _ := toStringArray(someTypes)
+		conceptType := getLeafType(stringTypes)
+		if conceptType != CONCEPT_TYPE_SPECIAL_REPORT && conceptType != CONCEPT_TYPE_SUBJECT {
+			if predicate == PREDICATE_IS_CLASSIFIED_BY && (conceptType == CONCEPT_TYPE_TOPIC || conceptType == CONCEPT_TYPE_LOCATION) {
+				originalAnnotations[i]["predicate"] = PREDICATE_ABOUT
+			} else if predicate == PREDICATE_IS_PRIMARILY_CLASSIFIED_BY && (conceptType == CONCEPT_TYPE_TOPIC || conceptType == CONCEPT_TYPE_LOCATION) {
+				originalAnnotations[i]["predicate"] = PREDICATE_ABOUT
+			} else if predicate == PREDICATE_MAJOR_MENTIONS {
+				originalAnnotations[i]["predicate"] = PREDICATE_ABOUT
+			}
```
Comment (galiri) [inline mapper/mapper.go]: as they all convert to ABOUT it could be just if with 3 or clauses

File: mapper/mapper_test.go [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+	expectedBody, err := ioutil.ReadFile("test_resources/annotations_isClassifiedBy_PAC.json")
+
+	if err != nil {
+		panic(err)
+	}
+	actualBody, _ := ConvertPredicates(originalBody)
+
+	log.Info(string(actualBody))
+	log.Info(string(expectedBody))
+
```
Comment (galiri) [inline mapper/mapper_test.go]: I'm wondering if we need to log this every time?

File: mapper/mapper_test.go [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+func TestIsPrimariTestConnectionErrorlyClassifiedByMappedCorrectly(t *testing.T) {
+
+	originalBody, err := ioutil.ReadFile("test_resources/annotations_isPrimarilyClassifiedBy_v2.json")
+	if err != nil {
+		panic(err)
+	}
+	expectedBody, err := ioutil.ReadFile("test_resources/annotations_isPrimarilyClassifiedBy_PAC.json")
+
```
Comment (galiri) [inline mapper/mapper_test.go]: Are we keeping isPrimarilyClassified as an annotation for types that are not Topics or Locations ?

File: mapper/mapper_test.go (hunk 3 above)
Comment (yolandeleungFT) [reply mapper/mapper_test.go]: I left that in by accident

File: api/api.yml [hunk 5]
```diff
@@ -41,7 +41,7 @@ paths:
         400:
           description: Invalid uuid supplied
         404:
-          description: Content not found
+          description: Annotations not found
```
[the remark this answers is not in the document]
Comment (yolandeleungFT) [reply api/api.yml]: We can't actually tell the difference between the content not found and no annotations found when we query neo for the relationship in public-annotations-api

**Pull request merged**